src/core_plugins/kibana/public/dashboard/panel/panel_header/panel_options_menu_container.js

```diff
@@ -0,0 +1,32 @@
+import { connect } from 'react-redux';
+
+import { PanelOptionsMenu } from './panel_options_menu';
+
+import {
+  deletePanel,
+  destroyEmbeddable
+} from '../../actions';
+
+import {
+  getEmbeddable,
+  getEmbeddableEditUrl,
+} from '../../reducers';
+
+const mapStateToProps = ({ dashboard }, { panelId }) => {
+  const embeddable = getEmbeddable(dashboard, panelId);
+  return {
+    editUrl: embeddable ? getEmbeddableEditUrl(dashboard, panelId) : '',
```

Comment:
FWIW, I don't think we need to create selectors for `getEmbeddableTitle`, `getEmbeddableEditUrl`, or `getEmbeddableError` -- it feels a little like overkill since at this point we're not really dealing with what I think of as "application state". I think it's enough to get the embeddable object, and just grab properties off it from wherever it's used.

Reply:
I was thinking this too when I saw the earlier PR that created all these selectors. I think it's totally acceptable to have something like `const embeddable = getEmbeddable(state, id)` and expect `embeddable` to be an object with whatever properties you want about it (title, edit url, etc). It creates extra code and boilerplate.  You can see this in `panel_header_container.js`, where you use `getEmbeddable` and then also use `getEmbeddableTitle`, two calls to Redux state, even though I'm guessing `getEmbeddable` gives you an object with the title already (or null).

That said, I don't think there's anything *wrong* with doing it this way. I wouldn't, but maybe I'm just lazy.

Reply:
I think this goes into the broader redux/selectors discussion. You both have valid points, but because there isn't consensus on where to draw this line, I'd prefer to just leave as is for now.  I don't even have a strong personal opinion, I just want to avoid changing code from A to B and back to B again depending on who happens to be reviewing the code.

Hopefully as teams continue to explore redux, more formality around these points will be formed and we can work them into a style guide.

Reply:
This seems like something we can't reach consensus on, it kind of comes down to personal choice. You have some arbitrary data, do you ask for objects and read properties off those objects, or do you ask for specific things about an object?

Reply:
After thinking about this some more, and having a good chat @kobelb and @chrisronline, I think I did take the selectors conversion too far. I will walk it back a bit, but will do so in a separate, focused PR.  I mean, how many times can I fit the word `Embeddables` on one line?  Apparently a lot! `export const getEmbeddableError =
  (dashboard, panelId) => getEmbeddableErrorFromEmbeddables(getEmbeddables(dashboard), panelId);` 😄

Reply:
++ agreed on this. I think it's about keeping the objects "fairly small", not going all the way to needing selectors for each field.